tree.go

```diff
@@ -1772,17 +1783,55 @@ func (n *LeafNode) serializeLeafWithUncompressedCommitments(cBytes, c1Bytes, c2B
 				children = append(children, padding...)
 			}
 		}
+
+		if isEoA {
+			switch i {
+			case 0:
+				isEoA = v != nil && bytes.Equal(v, zero32[:])
+			case 1:
+				// Balance should not be nil
+				isEoA = v != nil
+			case 2:
+				// last 24 bytes should be 0
+				isEoA = v != nil && bytes.Equal(v[8:32], zero24[:])
+			case 3:
+				// code hash should be the empty code hash, but let's be tolerant
+				// and accept 0 as well.
+				isEoA = v != nil && (bytes.Equal(v, emptyCodeHash[:]) || bytes.Equal(v, zero32[:]))
```

Comment:
Not sure we can be tolerant with zero here, since in the parsing method we hard code an `emptyCodeHash` value for `v[3]`.

Reply:
yeah I am reverting this to find out if this is what caused the issue I'm looking into.